### DISCUSSION TITLE
handling multiple terms which may occur within different files in the same project

### DISCUSSION THREAD

**Discussion by yaahc** [community/community#12834] [Other Feature Feedback, Questions, & Ideas] [answered]
I'm interested in using github code search to help find usages of APIs we're stabilizing in the rust standard library: example https://github.com/rust-lang/rust/issues/88581#issuecomment-1054642118

The problem I'm having right now though is I need to search for two different search terms which are likely to occur in different files. In the above example I handled it by searching for the first term, then individually searching within the projects that came back for the second term. I've not been able to figure out a way to concisely express this given the current API. The best way I can think of to solve this is to add a new scope option to the drop down next to the search bar after you've started a search.

![image](https://user-images.githubusercontent.com/1993852/158234957-31084a17-ba34-448e-a85e-cac7b02cd9a6.png)

Instead of only "all repos" or each specific repo that matched the search terms, I'd like an option for "all repos from the current search" to narrow the scope you're searching based on a previous search from which you can execute a subsequent search.

**Comment by colinwm** [comment, marked as answer]
I think I see what you're saying - you want to do something like: "find all repositories that have ... in their Cargo.toml", then take that set and search for usages of a feature somewhere else in that set of repos.

Unfortunately such a query is not possible to express right now. We're working on releasing an API though, which I think would be the answer for queries like this. You could make several queries, one to get repositories and another to search within them. The API would also allow you to exhaustively iterate through all of the results, which would probably be necessary for a search like that.

**Reply by yaahc** [reply]
> "find all repositories that have ... in their Cargo.toml"

In this case the info wouldn't be in the `Cargo.toml`, because I'm talking about standard library features, which end up needing a crate wide attribute usually in the `lib.rs` file like `#![feature(int_roundings)]`, but otherwise it sounds like you got the idea. The API you described sounds perfect! I look forward to when it's made available.
